src/test/java/org/broadinstitute/hellbender/engine/dataflow/ReadsSourceTest.java

```diff
@@ -0,0 +1,49 @@
+package org.broadinstitute.hellbender.engine.dataflow;
+
+import com.google.api.services.genomics.model.Read;
+import com.google.cloud.dataflow.sdk.Pipeline;
+import com.google.cloud.dataflow.sdk.testing.DataflowAssert;
+import com.google.cloud.dataflow.sdk.testing.TestPipeline;
+import com.google.cloud.dataflow.sdk.transforms.Count;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import com.google.cloud.genomics.dataflow.coders.GenericJsonCoder;
+import com.google.cloud.genomics.dataflow.utils.DataflowWorkarounds;
+import com.google.common.collect.ImmutableList;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.File;
+
+public class ReadsSourceTest extends BaseTest {
+    private final File bam = new File(this.getToolTestDataDir(), "count_reads_sorted.bam");
+
+    @Test
+    public void testLocal(){
```

Comment:
This should be named `testGetHeaderFromLocalBAM` instead of just `testLocal`, since it only looks at the header.